fathom/src/core/semantics.rs

```diff
@@ -598,6 +598,22 @@ fn prim_step(prim: Prim) -> Option<PrimStep> {
             })
         }
 
+        Prim::Array8Index | Prim::Array16Index | Prim::Array32Index | Prim::Array64Index => {
+            step!(context, [_, _, index, array] => match array.as_ref() {
+                Value::ArrayLit(elems) => {
+                    let index = match (index).as_ref() {
+                        Value::ConstLit(Const::U8(index, _)) => Some(usize::from(*index)),
+                        Value::ConstLit(Const::U16(index, _)) => Some(usize::from(*index)),
+                        Value::ConstLit(Const::U32(index, _)) => usize::try_from(*index).ok(),
+                        Value::ConstLit(Const::U64(index, _)) => usize::try_from(*index).ok(),
+                        _ => return None,
+                    }?;
+                    elems[index].clone()
```

Comment:
To be consistent with the other fallible ops I think we could change this to `elems.get(index).cloned()` - there's tradeoffs here but I think it’s probably better to avoid crashes in the evaluator that aren’t as a result of bugs in Fathom. 🤔